scripts/azure/SharePoint/Reset-SPOFolderAndFilesPermissions.ps1

```
@@ -1,90 +1,447 @@
+    $isDryRun = [bool]$a.IsDryRun
+
+    New-Item -ItemType Directory -Force -Path $out | Out-Null
+    $ts = Get-Date -Format "yyyyMMdd-HHmmss"
+    $flagCsv   = Join-Path $out "permissions-flagged-items-$ts.csv"
+    $countCsv  = Join-Path $out "folder-counts-$ts.csv"
+    $summaryJs = Join-Path $out "summary-$ts.json"
+
+    $worker.ReportProgress(0, @{type="log"; msg="[INFO] Connecting to SharePoint..."})
+
+    $sharingCapability = "Unknown"
+    $secure = ConvertTo-SecureString -String $pass -AsPlainText -Force
```

PSScriptAnalyzer (1.24.0)

[error] File 'Reset-SPOFolderAndFilesPermissions.ps1' uses ConvertTo-SecureString with plaintext. This will expose secure information. Encrypted standard strings should be used instead.

(PSAvoidUsingConvertToSecureStringWithPlainText)